modules/tensor_mechanics/src/materials/ADViscoplasticityStressUpdate.C

```diff
@@ -0,0 +1,290 @@
+{
+  // Compute the derivative of the gauge stress with respect to the equilvalent and hydrostatic
+  // stress components
+  const ADReal M = std::abs(_hydro_stress) / gauge_stress;
+  const ADReal h = computeH(_power, M);
+
+  // Compute the derviative of the residual with respect to the hydrostatic stress
+  ADReal dresidual_dhydro_stress = 0.0;
+  if (_hydro_stress)
+  {
+    const ADReal dM_dhydro_stress = M / _hydro_stress;
+    if (_model == ViscoplasticityModel::GTN)
+    {
+      dresidual_dhydro_stress = 2.0 * _intermediate_porosity * std::sinh(_pore_shape_factor * M) *
+                                _pore_shape_factor * dM_dhydro_stress;
+    }
+    else
+    {
+      const ADReal dresidual_dh =
+          _intermediate_porosity * (1.0 - _power_factor / Utility::pow<2>(h));
+      const ADReal dh_dM = computeH(_power, M, true);
+      dresidual_dhydro_stress = dresidual_dh * dh_dM * dM_dhydro_stress;
+    }
+  }
+
+  // Compute the derivative of the residual with respect to the equilvalent stress
+  ADReal dresidual_dequiv_stress = 2.0 * equiv_stress / Utility::pow<2>(gauge_stress);
+  if (_pore_shape == PoreShapeModel::SPHERICAL)
+    dresidual_dequiv_stress *= 1.0 + _intermediate_porosity / 1.5;
+
+  // Compute the derivative of the equilvalent stress to the deviatoric stress
+  const ADRankTwoTensor dequiv_stress_dsigma = 1.5 * dev_stress / equiv_stress;
+
+  // Compute the derivative of the residual with the stress
+  const ADRankTwoTensor dresidual_dsigma = dresidual_dhydro_stress * _dhydro_stress_dsigma +
+                                           dresidual_dequiv_stress * dequiv_stress_dsigma;
+
+  // Compute the deritative of the gauge stress with respect to the stress
+  const ADRankTwoTensor dgauge_dsigma =
+      gauge_stress * dresidual_dsigma / dresidual_dsigma.doubleContraction(stress);
+
+  return dgauge_dsigma;
+}
+
+template <ComputeStage compute_stage>
+void
+ADViscoplasticityStressUpdate<compute_stage>::computeInelasticStrainIncrement(
+    ADReal & gauge_stress,
+    ADReal & dpsi_dgauge,
+    ADRankTwoTensor & inelastic_strain_increment,
+    const ADReal & equiv_stress,
+    const ADRankTwoTensor & dev_stress,
+    const ADRankTwoTensor & stress)
+{
+  // If hydrostatic stress and porosity present, compute non-linear gauge stress
+  if (!_intermediate_porosity)
+    gauge_stress = equiv_stress;
+  else if (_hydro_stress)
+    returnMappingSolve(equiv_stress, gauge_stress, _console);
+  else
+    gauge_stress /= std::sqrt(1.0 + _power_factor * Utility::pow<2>(_intermediate_porosity));
+
+  mooseAssert(gauge_stress > equiv_stress,
+              "Gauge stress calculated in inner Newton solve is less than equilvalent.");
+
+  // Compute stress potential
+  dpsi_dgauge = _coefficient[_qp] * std::pow(gauge_stress, _power);
+
+  // Compute strain increment from stress potential and the gauge stress derivative with respect
+  // to the stress stress. The current form is explicit, and should eventually be changed
```

Comment:
It'd be good to add a small note to the documentation telling the reader that you've used explicit timestepping here.

Reply:
It is in `ADViscoplasticityStressUpdate.md`: `The inelastic strain is currently calculated explicitly,`